pkg/storage/streams/store.go

```diff
@@ -222,38 +210,37 @@ func (s *streamStore) Put(ctx context.Context, path paths.Path, data io.Reader,
 			if err != nil {
 				return nil, nil, err
 			}
-
+			encryptedStreamInfo, err := cipher.Encrypt(streamInfo, &encKey, &nonce)
```

Comment:
How is this nonce being generated? Is it being generated in the same way when we decrypt stream info?